nptdms/writer.py

```diff
@@ -17,11 +20,35 @@ class TdmsWriter(object):
             tdms_writer.write_segment(segment_data)
     """
 
-    def __init__(self, file, mode='w', version=4712):
+    @classmethod
+    def resave(cls, source, destination, version=4712, with_index_file=True, store_streams=True):
```

Comment:
I'm not sure about the `resave` name. Possibly `defragment` would be clearer?

And what's the point of the `store_streams` parameter? If someone wants to access the files for reading after, they should just open the files themselves, I don't think that functionality really belongs in the `TdmsWriter`.

The function documentation below is also missing documentation for the `with_index_file` and `store_streams` parameters, and the documentation for the `destination` parameter has been truncated.

Reply:
The self.streams dictionary consists of key `.tdms` if `iwth_index_file` is `False` and keys `.tdm` and `.tdms_index` with the corresponding streams. If not a file system path is supplied only the .tdms content is written to the stream and I need a possibility to access the stream for the `.tdms_index` as well. That way was the shorted that came into my mind. It would also raise an ValueError is value is set to true and a filepath is supplied for `destination`.
The documentation for the parameter is added and should make it a bit clearer. What do you think about this approach?

Reply:
Ah right thanks, I see now. It might be better if the user could pass in the index file as a parameter, then they can choose to use an on-disk file they've opened or use anything else that acts like a file, rather than needing to handle the BytesIO stream.

Eg. `TdmsWriter.__init__` could look like `__init__(self, file, mode='w', version=4712, with_index_file=False, index_file=None)`, and if `with_index_file` is True and file isn't a file path, then the `index_file` parameter needs to be provided. Although that's a bit awkward that there are two parameters that are closely related and you'd need to raise an error if `index_file` was passed but `with_index_file` was False. So maybe a better API would be `__init__(self, file, mode='w', version=4712, index_file=False)`, where the `index_file` can either be a boolean, and if True then the file parameter needs to be a path and we work out what the index file path should be, or `index_file` can be an opened file.

Reply:
That sounds like a good idea. I will implement it that way.